### PR TITLE
fix(gradle): Dynamically resolve `collectModulesScript` path

### LINKED ISSUE #2740: Make collecting modules on Android mono repo friendly
### Description

That means the path to `tools/collectModules.js` should be resolved relative to the plugin file path and not to the React root path.

https://github.com/getsentry/sentry-react-native/blob/8fbf60092842ca7f698340215dab1be0f53a29e6/sentry.gradle#L168

Currently, users can set their own path, but it can be solved out of the box.
https://github.com/getsentry/sentry-react-native/blob/8fbf60092842ca7f698340215dab1be0f53a29e6/sentry.gradle#L167

Based on https://github.com/getsentry/sentry-react-native/issues/2702

### PR DESCRIPTION
**Pull request by krystofwoldrich**
## :loudspeaker: Type of change

- [x] Enhancement

## :scroll: Description

Thank @mccraveiro for the original PR, I've opened this one for the e2e CI to run to see that the change is working.

Current script path resolution expects the files to be located in the project root in node_modules which doesn't work for monorepo setups. This PR adds dynamic resolve of the sentry package dir to work with monorepos.

## :bulb: Motivation and Context

When using Sentry on monorepos we need to manually set the collectModulesScript path
Original PR https://github.com/getsentry/sentry-react-native/pull/3070
Closes #2740

## :green_heart: How did you test it?
CI - Both Android RN [0.65](https://sentry-sdks.sentry.io/issues/4045737360/events/dced9978f27d442faa6d54a99f825653/) and [0.71](https://sentry-sdks.sentry.io/issues/4045737360/events/e47a4a7a43ed4743b6a6bff8b2466b08/) are sending events with modules.

## :pencil: Checklist

- [x] I reviewed submitted code
- [ ] I added tests to verify changes
- [x] No new PII added or SDK only sends newly added PII if `sendDefaultPII` is enabled
- [x] All tests passing
- [x] No breaking changes

## :crystal_ball: Next steps

### CODE REVIEW

**Comment by krystofwoldrich** [comment]
[Android 0.65 with modules](https://sentry-sdks.sentry.io/share/issue/05ceb536912d4b18915954b06ae99298/)
[Android 0.71 with modules](https://sentry-sdks.sentry.io/share/issue/05ceb536912d4b18915954b06ae99298/)

File: sentry.gradle [hunk 1]
```diff
@@ -166,9 +166,15 @@ gradle.projectsEvaluated {
 
               workingDir reactRoot
 
+              def resolvedSentryPath = null
+              try {
+                  resolvedSentryPath = new File(["node", "--print", "require.resolve('@sentry/react-native/package.json')"].execute(null, rootDir).text.trim()).getParentFile();
+              } catch (Throwable ignored) {}
```
Comment (marandaneto) [inline sentry.gradle:172]: L: Add a comment to explain why it is ignored.

File: CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,10 @@
 
 ## Unreleased
 
+### Features
```
Comment (marandaneto) [inline CHANGELOG.md:5]: L: Would that be a fix instead?

**Review by marandaneto — APPROVED** [review]

**Comment by taylorkline** [comment]
Thank you!

**Pull request merged**